gem/ci/build.ps1

```diff
@@ -31,9 +31,13 @@ function Lint-Dockerfile(
         $ignores += '--ignore', $code
     }
 
-    Write-Output "hadolint $ignores $Path"
+    Write-Host "& cmd.exe /c docker run --rm -i hadolint/hadolint hadolint $ignores -< $Path"
 
-    hadolint $ignores $Path
+    # while a simpler method works locally, there appears to be a bug with stdin in Azure injecting BOMs?
+    # https://developercommunity.visualstudio.com/content/problem/451239/powershell-build-step-in-azure-devops-fails-when-p.html
+    # https://github.com/microsoft/botbuilder-tools/pull/1046
+    & cmd.exe /c docker run --rm -i hadolint/hadolint hadolint $ignores -< $Path
```

Comment:
FWIW we volume-map in the Dockerfile in our Makefiles, so we don't use stdin in that case.

```
	@docker pull hadolint/hadolint:latest
	@docker run --rm -v $(PWD)/pe-client-tools/Dockerfile:/Dockerfile \
		-i hadolint/hadolint:latest hadolint --ignore DL3008 --ignore DL3018 --ignore DL4000 --ignore DL4001 Dockerfile
```